### PR TITLE
Nc news 9

### PR DESCRIPTION
**Pull request by jakejones2**

### CODE REVIEW

File: __tests__/comments.test.js [hunk 1]
```diff
@@ -198,3 +198,39 @@ describe("POST comments", () => {
...
 });
+
+describe("DELETE comments by id", () => {
+  test("DELETE 204 from /api/comments/:comment_id", () => {
+    return request(app).delete("/api/comments/1").expect(204);
+  });
+  test("DELETE comments deletes comment", () => {
+    return request(app)
+      .delete("/api/comments/1")
+      .expect(204)
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/comments.test.js]: can remove `expect(204)` from this test as it's covered in the previous test

File: __tests__/comments.test.js [hunk 2]
```diff
@@ -198,3 +198,39 @@ describe("POST comments", () => {
...
+      .then(() => {
+        return request(app)
+          .get("/api/articles/9/comments")
+          .expect(200)
+          .then(({ body: { comments } }) => {
+            // used to be two comments, see data
+            expect(comments.length).toBe(1);
+          });
+      });
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/comments.test.js:217]: This is a nifty way to check it's actually deleted the comment!

File: __tests__/comments.test.js [hunk 3]
```diff
@@ -198,3 +198,39 @@ describe("POST comments", () => {
...
+  });
+  test("Returns 404 if comment not found", () => {
+    return request(app)
+      .delete("/api/comments/1000")
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Comment not found");
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/comments.test.js:224]: Really good error handling to give a clear and specific message about the comment not being found!

File: __tests__/comments.test.js [hunk 4]
```diff
@@ -198,3 +198,39 @@ describe("POST comments", () => {
...
+      });
+  });
+  test("Returns 400 if comment_id invalid", () => {
+    return request(app)
+      .delete("/api/comments/fryingpan")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad Request");
+      });
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/comments.test.js:233]: Good stuff

File: controllers/comments-controller.js [hunk 5]
```diff
@@ -27,4 +31,14 @@ function createComments(req, res, next) {
     });
 }
 
-module.exports = { getComments, createComments };
+function deleteComment(req, res, next) {
+  removeComment(req.params.comment_id)
+    .then(() => {
+      res.status(204).send();
```
Comment (Pipe-Smoking-Rabbit) [inline controllers/comments-controller.js]: some syntactic sugar for an empty body: `res.sendStatus(204)` 🪄

File: controllers/comments-controller.js [hunk 6]
```diff
@@ -27,4 +31,14 @@ function createComments(req, res, next) {
...
+    })
+    .catch((err) => {
+      next(err);
+    });
```
Comment (Pipe-Smoking-Rabbit) [inline controllers/comments-controller.js]: some more syntactic sugar, if you only want to invoke `next` in the catch block: 
`.catch(next);` - express will know what to do and passes the error back to app 🪄

**Review by Pipe-Smoking-Rabbit — APPROVED** [review]
Really good clean code on this ticket!

File: controllers/comments-controller.js (hunk 6 above)
Comment (jakejones2) [reply controllers/comments-controller.js]: thanks for all the refactor tips! Made these changes :)

**Pull request merged**